zebra-network/src/meta_addr.rs

```diff
@@ -263,6 +264,76 @@ impl ZcashDeserialize for MetaAddr {
         Ok(MetaAddr::new_gossiped(&addr, &services, &last_seen))
     }
 }
+/// A serialized meta addr has a 4 byte time, 8 byte services, 16 byte IP addr, and 2 byte port
+const META_ADDR_SIZE: usize = 4 + 8 + 16 + 2;
+impl TrustedPreallocate for MetaAddr {
+    fn max_allocation() -> u64 {
+        // Since a maximal serialized Vec<MetAddr> uses at least three bytes for its length (2MB  messages / 30B MetaAddr implies the maximal length is much greater than 253)
+        // the max allocation can never exceed (MAX_PROTOCOL_MESSAGE_LEN - 3) / META_ADDR_SIZE
+        ((MAX_PROTOCOL_MESSAGE_LEN - 3) / META_ADDR_SIZE) as u64
+    }
+}
+
+#[cfg(test)]
+mod test_trusted_preallocate {
+    use std::convert::TryInto;
+
+    use super::{MetaAddr, MAX_PROTOCOL_MESSAGE_LEN, META_ADDR_SIZE};
+    use super::{PeerAddrState, PeerServices};
+    use chrono::{TimeZone, Utc};
+    use zebra_chain::serialization::{TrustedPreallocate, ZcashSerialize};
+    #[test]
+    /// Confirm that each MetaAddr takes exactly META_ADDR_SIZE bytes when serialized.
+    /// This verifies that our calculated `TrustedPreallocate::max_allocation()` is indeed an upper bound.
+    fn meta_addr_size_is_correct() {
```

Comment:
The `MetaAddr` tests should be proptests, but I can make that change in a follow-up PR.

Reply:
(In case we add more address types in future, for example, Tor onion addresses.)